schema.ts

```diff
@@ -1232,6 +1232,43 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
 }
 
+/**
+ * Represents a permission in the external system
+ *
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+export interface Permission {
+  /**
+   * Indicates whether the permission is public to all users who have access to the ingestion
+   */
+  isPublic: boolean;
+
+  /**
+   * The ID of the user associated with the permission
+   *
+   * This must match the ids returned from {@link BaseAuthentication.getConnectionUserId}
+   * or the ids specified in the {@link ObjectSchemaDefinition.userIdProperty} column of a table
+   * with role {@link TableRole.Users}
+   */
+  userId: string;
+
+  /**
+   * The ID of the group associated with the permission.
+   *
+   * TODO(sam): Update with description of where this group id is set
+   */
+  groupId: string;
+
+  /**
+   * The domain name associated with the permission.
+   */
+  domainName: string;
```

Comment:
Do we need this?  Can we look this up based on the group?

Reply:
We need this. Domain Name is a separate auth mechanism then groups. We can potentially remove it for v0 though as we may bring in support later down the line

Reply:
So then should groupId and domainName be `| undefined` (or `?`)?  And then we document that at least one must be specified?  Can both be specified?  Should we use an additional type to make these things more clear?

Reply:
I guess I'm thinking about this from a non-Coda perspective.  What if someone wants some data to be open to the world, some to be limited by domain, and some to be limited by group, how would they do that?  And do the comments provide enough context to implement what they want.

Reply:
Good point these should all be optional and at most 1 should be specified per row